File: dotnet/mission-to-mars/Helicoptere.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace mission_to_mars
{
    public class Helicoptere : Rover, IModuleMartien
    {
        public Helicoptere(Direction direction, Position position) : base(direction, position)
        {
        }

        public override void Monter()
        {
            Position = Position with { Z = Position.Z + 1 };
        }


        public override void Descendre()
        {
            Position = Position with { Z = Position.Z - 1 };
        }
    }
}

File: typescript/src/Helicoptere.ts
import { Direction } from './Direction'
import { ModuleMartien } from './ModuleMartien'
import { Position } from './Position'
import { Rover } from './Rover'

export class Helicoptere extends Rover implements ModuleMartien {
  constructor(override readonly direction: Direction, override readonly position: Position) {
    super(direction, position);
  }

  override monter() {
    super._position = new Position(this._position.x, this._position.y, this._position.z + 1);
  }

  override descendre() {
    super._position = new Position(this._position.x, this._position.y, this._position.z - 1);
  }
}
